File: agent-orchestrator-mcp-server/src/logger.ts
/**
 * Simple file-based logger for debugging MCP server
 */

import * as fs from "fs";
import * as path from "path";
import { fileURLToPath } from "url";

const __filename = fileURLToPath(import.meta.url);
const __dirname = path.dirname(__filename);

// Create logs directory relative to the MCP server root
const LOG_DIR = path.join(__dirname, "..", "logs");
const LOG_FILE = path.join(LOG_DIR, "mcp-server.log");

// Ensure logs directory exists
function ensureLogDir() {
  if (!fs.existsSync(LOG_DIR)) {
    fs.mkdirSync(LOG_DIR, { recursive: true });
  }
}

/**
 * Log levels
 */
export enum LogLevel {
  DEBUG = "DEBUG",
  INFO = "INFO",
  WARN = "WARN",
  ERROR = "ERROR"
}

/**
 * Check if logging is enabled via environment variable
 */
function isLoggingEnabled(): boolean {
  return process.env.MCP_SERVER_DEBUG === "true";
}

/**
 * Write a log entry to the log file
 */
export function log(level: LogLevel, message: string, data?: any) {
  // Only log if debugging is enabled
  if (!isLoggingEnabled()) {
    return;
  }

  try {
    ensureLogDir();

    const timestamp = new Date().toISOString();
    const logEntry = {
      timestamp,
      level,
      message,
      ...(data && { data })
    };

    const logLine = JSON.stringify(logEntry) + "\n";

    fs.appendFileSync(LOG_FILE, logLine);
  } catch (error) {
    // If logging fails, write to stderr but don't crash
    console.error("Failed to write to log file:", error);
  }
}

/**
 * Convenience methods
 */
export const logger = {
  debug: (message: string, data?: any) => log(LogLevel.DEBUG, message, data),
  info: (message: string, data?: any) => log(LogLevel.INFO, message, data),
  warn: (message: string, data?: any) => log(LogLevel.WARN, message, data),
  error: (message: string, data?: any) => log(LogLevel.ERROR, message, data),
};

/**
 * Clear the log file (useful for testing)
 */
export function clearLog() {
  if (!isLoggingEnabled()) {
    return;
  }

  try {
    ensureLogDir();
    fs.writeFileSync(LOG_FILE, "");
  } catch (error) {
    console.error("Failed to clear log file:", error);
  }
}

File: agent-orchestrator/skills/agent-orchestrator/mcp-server/libs/logger.py
"""
Simple file-based logger for debugging MCP server
"""

import json
import os
from datetime import datetime
from enum import Enum
from pathlib import Path
from typing import Any, Optional


class LogLevel(str, Enum):
    """Log levels"""
    DEBUG = "DEBUG"
    INFO = "INFO"
    WARN = "WARN"
    ERROR = "ERROR"


# Get the project root directory (parent of src/)
PROJECT_ROOT = Path(__file__).parent.parent.parent
LOG_DIR = PROJECT_ROOT / "logs"
LOG_FILE = LOG_DIR / "mcp-server.log"


def is_logging_enabled() -> bool:
    """Check if logging is enabled via environment variable"""
    return os.environ.get("MCP_SERVER_DEBUG", "false").lower() == "true"


def ensure_log_dir() -> None:
    """Ensure logs directory exists"""
    if not LOG_DIR.exists():
        LOG_DIR.mkdir(parents=True, exist_ok=True)


def log(level: LogLevel, message: str, data: Optional[Any] = None) -> None:
    """Write a log entry to the log file"""
    # Only log if debugging is enabled
    if not is_logging_enabled():
        return

    try:
        ensure_log_dir()

        timestamp = datetime.utcnow().isoformat() + "Z"
        log_entry = {
            "timestamp": timestamp,
            "level": level.value,
            "message": message,
        }

        if data is not None:
            log_entry["data"] = data

        log_line = json.dumps(log_entry) + "\n"

        with open(LOG_FILE, "a") as f:
            f.write(log_line)

    except Exception as error:
        # If logging fails, write to stderr but don't crash
        print(f"Failed to write to log file: {error}", file=__import__("sys").stderr)


class Logger:
    """Logger convenience class"""

    @staticmethod
    def debug(message: str, data: Optional[Any] = None) -> None:
        log(LogLevel.DEBUG, message, data)

    @staticmethod
    def info(message: str, data: Optional[Any] = None) -> None:
        log(LogLevel.INFO, message, data)

    @staticmethod
    def warn(message: str, data: Optional[Any] = None) -> None:
        log(LogLevel.WARN, message, data)

    @staticmethod
    def error(message: str, data: Optional[Any] = None) -> None:
        log(LogLevel.ERROR, message, data)


# Export singleton logger instance
logger = Logger()


def clear_log() -> None:
    """Clear the log file (useful for testing)"""
    if not is_logging_enabled():
        return

    try:
        ensure_log_dir()
        with open(LOG_FILE, "w") as f:
            f.write("")
    except Exception as error:
        print(f"Failed to clear log file: {error}", file=__import__("sys").stderr)
